### PR TITLE
watchman 4.6.0-rc2 (devel)

### PR DESCRIPTION
**Pull request by ilovezfs**
- add devel spec
- make test work for both stable and devel
- set statedir to a directory outside of the prefix

### CODE REVIEW

File: Formula/watchman.rb [hunk 1]
```diff
@@ -23,7 +29,8 @@ def install
                           "--with-pcre",
                           # we'll do the homebrew specific python
                           # installation below
-                          "--without-python"
+                          "--without-python",
+                          "--enable-statedir=#{var}/run/watchman"
```
Comment (DomT4) [inline Formula/watchman.rb]: We'll want to make sure this is created in `post_install`.

**Comment by DomT4** [comment]
Single nit but 👍 otherwise.

File: Formula/watchman.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/watchman.rb]: Seems to create itself on demand as needed ...

Comment (ilovezfs) [inline Formula/watchman.rb]: Actually, it's created by the formula at build time. So not sure what to do about that. If `/usr/local/var` exists at the beginning and it creates `/usr/local/var/run` and `/usr/local/var/run/watchman` during the build, do we then need to `rm -rf /usr/local/var/run` at the end of the build in `install`, and then `mkdir_p /usr/local/var/run/watchman` during `post_install`? But if at the beginning `/usr/local/var/run` already existed, and during `install` the build only creates `/usr/local/var/run/watchman`, then do we need to `rm -rf  /usr/local/var/run/watchman` only and then during post_install `mkdir /usr/local/var/run/watchman`?

Comment (DomT4) [inline Formula/watchman.rb]: It it's created during build that's pretty much fine. The problems tend to start when it isn't and then the executable is called with root, either via `sudo` or `launchctl` and that directory becomes unwritable for Homebrew.

Comment (ilovezfs) [inline Formula/watchman.rb]: This is what you get by the end of `install` if neither `/usr/local/var/run` nor `/usr/local/var/run/watchman` exist at the beginning:
```
$ ls -ld /usr/local/var/run
drwxr-xr-x  3 joe  admin  102 Jun  9 06:56 /usr/local/var/run
$ ls -ld /usr/local/var/run/watchman
drwxrwsrwx  2 joe  admin  102 Jun  9 06:56 /usr/local/var/run/watchman
```

Comment (DomT4) [inline Formula/watchman.rb]: Seems alright as-is then 👍.

Comment (ilovezfs) [inline Formula/watchman.rb]: I'll check to make sure it works via bottle install.

Comment (ilovezfs) [inline Formula/watchman.rb]: Interestingly the bottle screws up the permissions:
```
$ ls -ld /usr/local/var/run
drwxr-xr-x  3 joe  admin  102 Jun  9 07:06 /usr/local/var/run
$ ls -ld /usr/local/var/run/watchman
drwxr-xr-x  2 joe  admin  68 Jun  9 07:06 /usr/local/var/run/watchman
```

Comment (DomT4) [inline Formula/watchman.rb]: I suspect it's attempting to "be helpful".

Comment (ilovezfs) [inline Formula/watchman.rb]: I guess we want
```
  def post_install
    (var/"run/watchman").mkpath
    chmod "uo=rwx,g=rwxs", var/"run/watchman"
  end
```

Comment (ilovezfs) [inline Formula/watchman.rb]:
```
iMac-TMP:local joe$ brew install ./watchman-4.6.0-rc1.el_capitan.bottle.tar.gz 
==> Pouring watchman-4.6.0-rc1.el_capitan.bottle.tar.gz
🍺  /usr/local/Cellar/watchman/4.6.0-rc1: 21 files, 387.5K
iMac-TMP:local joe$ ls -ld /usr/local/var/run
drwxr-xr-x  3 joe  admin  102 Jun  9 07:23 /usr/local/var/run
iMac-TMP:local joe$ ls -ld /usr/local/var/run/watchman
drwxrwsrwx  2 joe  admin  68 Jun  9 07:23 /usr/local/var/run/watchman
```

Comment (ilovezfs) [inline Formula/watchman.rb]: OK, added a `post_install` method.

Comment (ilovezfs) [inline Formula/watchman.rb]: which of course causes CI failure.

Comment (ilovezfs) [inline Formula/watchman.rb]: OK, using 042777 instead of "uo=rwx,g=rwxs" should make it happy on Mavericks CI, too.

Comment (DomT4) [inline Formula/watchman.rb]: Smells like it's worth hunting why we change permissions on bottled `var` and whether doing so has any downsides.

Comment (ilovezfs) [inline Formula/watchman.rb]: I agree. Bottles should preserve the fidelity of the permissions of what's installed (of course, including whatever permissions "fixes" brew may have done during install).

**Pull request closed without merging**